LaunchPadBooster/Utils/InputLock.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Assets.Scripts;
+using Assets.Scripts.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LaunchPadBooster.Utils
+{
+    /// <summary>
+    /// Creating this object will create an input lock in the game.
+    /// This prevents keyboard or mouse input from doing what it normally does.
+    /// Cancelling the lock is done by calling Dispose() on the object.
+    /// </summary>
+    public class InputLock : IDisposable
+    {
+        [Flags]
+        public enum LockType : int
+        {
+            None = 0,
+            Keyboard = 1,
+            Mouse = 2,
+        }
+
+        /// <summary>
+        /// Creates a new input lock.
+        /// </summary>
+        /// <param name="lockType">What input to lock</param>
+        public InputLock(LockType lockType = LockType.Keyboard)
+        {
+            this.lockType = lockType;
+            lockName = $"LaunchPadBoosterLock-{lockCounter++}";
+
+            if (lockType.HasFlag(LockType.Keyboard))
+            {
+                KeyManager.SetInputState(lockName, KeyInputState.Typing);
+                CursorManager.SetCursor(true);
+                InputWindow.InputState = InputPanelState.Waiting;
+            }
+
+            if (lockType.HasFlag(LockType.Mouse))
+            {
+                InputMouse.SetMouseControl(true);
+            }
+        }
+
+        private bool disposedValue;
+        private string lockName;
+        private LockType lockType;
+
+        private static long lockCounter = 0;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    // There is no managed state (managed objects)
+                }
+
+                if (lockType.HasFlag(LockType.Keyboard))
+                {
+                    KeyManager.RemoveInputState(lockName);
+                    CursorManager.SetCursor(false);
+                    InputWindow.InputState = InputPanelState.None;
+                }
+
+                if (lockType.HasFlag(LockType.Mouse))
+                {
+                    InputMouse.SetMouseControl(false);
+                }
+                disposedValue = true;
+            }
+        }
+
+        ~InputLock()
```

Comment:
I don't think we want the full finalizer pattern here. if this is isn't properly called with a using block, there's no way to know when the finalizer will actually run, and its unlikely to fix the issues caused by calling it incorrectly. Should be able to just have a simple constructor and dispose

Reply:
The issue this guards against is programmer error, and if its being called in any way other than `using (new InputLock(...))`, then nothing we set up here can ensure the right behavior since we can't guarantee the specific time the finalizer is called. I think the added complexity isn't worth it